utils/queryApiGateway.ts

```diff
@@ -176,6 +176,16 @@ export default async function queryApiGateway(req: IncomingMessage, uniqueNumber
   } catch (error) {
     const logger: Logger = Logger.getInstance()
     logger.log('error', error, 'API gateway error')
+    throw error
+  }
+
+  if (apiData?.uniqueNumber !== uniqueNumber) {
+    const mismatchError = new Error(
+      `Mismatched API response and Header unique number (${apiData.uniqueNumber || 'null'} and ${uniqueNumber})`,
+    )
+    const logger: Logger = Logger.getInstance()
+    logger.log('error', mismatchError, 'Unexpected API gateway response')
+    throw mismatchError
```

Comment:
@lomky I hope you don't mind: I did a little refactor of your code here because we don't need to do a `try catch` here.